### DISCUSSION TITLE
open Web3NetworkSwitch with custom button

### DISCUSSION THREAD

**Discussion by JustAudrius** [WalletConnect/walletconnect-monorepo#2487] [Developer Support] [answered]
I want open modal from custom button.
Example:

```
import { useWeb3Modal } from "@web3modal/react";


const CustomButton = () => {
  const { open } = useWeb3Modal();
  const openModal = () => {
    open()
  }
  return (
    <>
      <button onClick={openModal}>Open network switch modal</button>
    </>
  );
};
```

but I want only with network switch not wallet modal.

Thank you.

**Comment by xzilja** [comment, marked as answer]
`open` function  takes in some options, in particular `route` which you can set to open network view
https://docs.walletconnect.com/2.0/web/web3modal/react/wagmi/hooks#reference
